client/state/themes/actions.js

```diff
@@ -423,8 +423,14 @@ export function installTheme( themeId, siteId ) {
 		} );
 
 		return wpcom.undocumented().installThemeOnJetpack( siteId, themeId )
-			.then( ( theme ) => {
-				dispatch( receiveTheme( theme, siteId ) );
+			.then( ( /* theme */ ) => {
+				// We do not `dispatch( receiveTheme( theme, siteId ) )` here because
+				// in our UI, themes from WP.com (and WP.org) are already present in
+				// a separate list, and we do not want to duplicate them.
+				// Arguably, this is too tightly coupled to our UI, but because pagination,
+				// it's easier to filter _before_ storing themes in ThemeQueryManager than
+				// when fetching themes from it.
```

Comment:
I'll remove the lower half, which is a bit over-specific here. The upper half I'd like to keep around just so no-one feels the urge to add that back in a couple of months.